python/private/pypi/whl_library_targets.bzl

```diff
@@ -0,0 +1,339 @@
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Macro to generate all of the targets present in a {obj}`whl_library`."""
+
+load("@bazel_skylib//rules:copy_file.bzl", "copy_file")
+load("//python:py_binary.bzl", "py_binary")
+load("//python:py_library.bzl", "py_library")
+load("//python/private:normalize_name.bzl", "normalize_name")
+load(
+    ":labels.bzl",
+    "DATA_LABEL",
+    "DIST_INFO_LABEL",
+    "PY_LIBRARY_IMPL_LABEL",
+    "PY_LIBRARY_PUBLIC_LABEL",
+    "WHEEL_ENTRY_POINT_PREFIX",
+    "WHEEL_FILE_IMPL_LABEL",
+    "WHEEL_FILE_PUBLIC_LABEL",
+)
+
+def whl_library_targets(
+        name,
+        *,
+        whl_name,
+        dep_template,
+        data_exclude = [],
+        srcs_exclude = [],
+        tags = [],
+        filegroups = {
+            DIST_INFO_LABEL: ["site-packages/*.dist-info/**"],
+            DATA_LABEL: ["data/**"],
```

Comment:
FYI, I would like to include extra folders here, what me and @arrdem discussed in #1730. It would be valuable to have those files so that the `hub` repo can provide more `filegroup` targets out of the box. This may make things easier where the `annotations` API is heavily used by our users.

cc: @keith, not sure if this help in defining the `cc_library` targets, but regardless, this may be a good step forward.